dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,258 @@
+            parser,
+            debug):
+        self.work_file = work_file
+        self.parser = parser
+        self.debug = debug
+
+    def remove_incorrect_works(self, work_listing):
+        """Remove works that have an incorrect structure in a list of works.
+
+        Return
+            A new list with the incorrect works removed."""
+
+        work_to_remove = []
+
+        logger.debug("Removing the incorrect work from the current work type.")
+        for dict_work in work_listing:
+
+            # check if it is a dictionary
+            if not isinstance(dict_work, dict):
+                work_to_remove.append(dict_work)
+                logger.debug("Incorrect work : value is not a dictionary.")
+                continue
+
+            # check if it has a title field
+            work_title = dict_work.get('title')
+            if not work_title:
+                work_to_remove.append(dict_work)
+                logger.debug("Incorrect work : no title field found")
+                continue
+
+            logger.debug("Work '{}' is correctly structured.".format(
+                    work_title))
+
+        return [work for work in work_listing if work not in work_to_remove]
+
+    def debug_parser_work(self, dict_work):
+        """Display debug messages on a specific work dictionarry.
+
+        Args
+            dict_work: dictionnary for which the check is done
+            (must have a 'title' field)
+        """
+        field_names = ('subtitle', 'alternative_titles')
+
+        # get the work title (should have been checked it exists
+        # before calling this method)
+        work_title = dict_work.get('title')
+
+        for field in dict_work:
+            if field not in field_names:
+                if work_title:
+                    logger.debug(
+                        "Field '{}' for '{}' not in fields taken"
+                        " into account.".format(
+                            field,
+                            work_title))
+                else:
+                    logger.debug(
+                        "Field '{}' not in fields taken"
+                        " into account.".format(field))
+
+    def creatework(self, work_type_entry, dict_work):
+        """Create or update a work in database."""
+
+        # get the attributes of the work
+        work_title = dict_work.get('title')
+        work_subtitle = dict_work.get('subtitle', "")
+        work_alternative_titles = dict_work.get('alternative_titles', [])
+
+        # get or create works
+        work_entry, work_created = Work.objects.get_or_create(
+            title__iexact=work_title,
+            work_type=work_type_entry,
+            subtitle__iexact=work_subtitle,
+            defaults={
+                'title': work_title,
+                'work_type': work_type_entry,
+                'subtitle': work_subtitle}
+            )
+
+        if work_created:
+            logger.debug("Created work '{}' subtitled '{}'.".format(
```

Comment:
You should use the Work object string method, which is designed for cases like this:
```python
"Created {}".format(work_entry)
```
However, the current string method does not display the subtitle. Since it is clear now that works are identified by both their title and subtitle, it could be worth to update [this method](https://github.com/DakaraProject/dakara-server/blob/b7b11cc0fd9605a4312d6ed36b20f97954b8d5b0/dakara_server/library/models.py#L49).

Reply:
The alternative title str seems to be also incorrect (not consistent with the others). Should we open an issue for both of them ?

Reply:
Do you mean not consistent for the formulation of the sentence? You can improve it if you want.

You can perform the changes in this PR directly.